### PR TITLE
DevOps: Added recipe for transaction sync testing.

### PR DESCRIPTION
**Pull request by egieseke**
## Summary

Added "txnsync" recipe matching configuration requirements for txn sync testing on Betanet. 

## Test Plan

Tested recipe by using algonet to launch a network.  Verified that the rounds were progressing.

### CODE REVIEW

**Review by algobarb — COMMENTED** [review]
Is this just for Phase0 testing and a Betanet replica? Or would this be the format used for the Betanet Deployment?
Betanet TxnSync deployement would need 4 betanet relays and 10 additional non-participation nodes using m5d.xlarge machines, geo-distributed. I see participation nodes and that all the nodes here are c5.4xlarge machines and in us-east-2. Can we clarify the purpose of the recipe?

**Comment by egieseke** [comment]
> Is this just for Phase0 testing and a Betanet replica? Or would this be the format used for the Betanet Deployment? Betanet TxnSync deployement would need 4 betanet relays and 10 additional non-participation nodes using m5d.xlarge machines, geo-distributed. I see participation nodes and that all the nodes here are c5.4xlarge machines and in us-east-2. Can we clarify the purpose of the recipe?

@algobarb This recipe is used to create the betanet network extension for the txn sync testing.  I used c5.4xlarge machines to match what was used for the current betanet relays and nodes based on _"Create R6..R9 on different geo-zones. Use the same host configuration used for the rest of the betanet relays."_ and _"Create 5 hosts, using the same host configuration as R5"_. We can easily change to md5.xlarge if those are required.  

I will update the recipe to spread out to different regions, good catch!

**Comment by algobarb** [comment]
>

**Comment by algobarb** [comment]
> > Is this just for Phase0 testing and a Betanet replica? Or would this be the format used for the Betanet Deployment? Betanet TxnSync deployement would need 4 betanet relays and 10 additional non-participation nodes using m5d.xlarge machines, geo-distributed. I see participation nodes and that all the nodes here are c5.4xlarge machines and in us-east-2. Can we clarify the purpose of the recipe?
> 
> @algobarb This recipe is used to create the betanet network extension for the txn sync testing. I used c5.4xlarge machines to match what was used for the current betanet relays and nodes based on _"Create R6..R9 on different geo-zones. Use the same host configuration used for the rest of the betanet relays."_ and _"Create 5 hosts, using the same host configuration as R5"_. We can easily change to md5.xlarge if those are required.
> 
> I will update the recipe to spread out to different regions, good catch!

"Expected cost: Use m5d.xlarge for each of the relays/nodes." under BetaNet Deployment in the Deployment doc. Host configuration is still the same, but machine type was supposed to be different. When we reviewed the documentation we settled on that we need a total of 9 machines: 4 machines with 1 Relay each, and then 5 machines with 2 NPN each. So we'll have to adjust this and maybe have a review with John since Tsachi is out `{"R":4, "N":5, "NPN":1}` 
It's nice that you've got it to work though so that adjusting the requirements would be much easier than if we had manually spun these up.

File: test/testdata/deployednettemplates/recipes/txnsync/topology.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      "Name": "R3",
+      "Template": "AWS-EU-WEST-2-Large"
+    },
+    {
+      "Name": "R4",
+      "Template": "AWS-EU-CENTRAL-1-Large"
+    },
+    {
+      "Name": "N1",
+      "Template": "AWS-US-EAST-2-Large"
```
Comment (onetechnical) [inline test/testdata/deployednettemplates/recipes/txnsync/topology.json:21]: This topology doesn't seem correct. Refer to the txync deployment document.

**Review by onetechnical — CHANGES_REQUESTED** [review]

File: test/testdata/deployednettemplates/recipes/txnsync/topology.json (hunk 1 above)
Comment (egieseke) [reply test/testdata/deployednettemplates/recipes/txnsync/topology.json:21]: Do you mean the names should be updated?  Or the Machine sizes?  Or Both?

Comment (onetechnical) [reply test/testdata/deployednettemplates/recipes/txnsync/topology.json:21]: Nope, this is correct, I'm mistaken

**Review by onetechnical — APPROVED** [review]

**Review by algobarb — APPROVED** [review]
Thanks for clarifying. I wasn't sure if there was a need for them to be NPN in the original doc.

**Review by algojack — APPROVED** [review]

**Pull request merged**